src/parseAndDecorate.ts

```diff
@@ -202,6 +196,15 @@ export class DecoratorAndParser {
 			}
 		}
 
+		function validateAndAddPlaceholder(part: string, partOffset: number, partOffsetEnd: number) {
+			if (placeholderNameRegex.exec(part) !== null) {
+				placeHoldersForKey.get(messageKey!)!.push(new Literal(part, partOffset, partOffsetEnd));
+				decorateAt(partOffset, partOffsetEnd, placeholderDecoration);
+			} else {
+				showErrorAt(partOffset, partOffsetEnd, `"${part}" is not a valid placeholder name.`, vscode.DiagnosticSeverity.Error);
```

Comment:
It might be useful to explain why it's not valid (for example it needs to be a valid Dart variable name)